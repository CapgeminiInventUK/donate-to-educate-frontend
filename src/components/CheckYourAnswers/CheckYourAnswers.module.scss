@use '@assets/global.scss' as g;

.container {
  color: g.$primary-dark-blue;
  margin-bottom: 60px;

  .row {
    font-size: 18px;
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 22px 0;
    border-bottom: 1px solid g.$primary-light-blue;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.04));
    line-height: 2;

    div {
      flex-basis: 100%;
    }

    .field {
      font-weight: 700;
    }

    .value {
      white-space: pre-line;
      font-weight: 400;
    }

    .addressCell {
      line-height: 1.5;
    }

    .changeLink {
      display: flex;
      justify-content: center;
      align-items: center;

      button {
        color: g.$secondary-mid-blue;
        font-weight: 400;
        text-underline-offset: 2px;
      }
    }

    .changeAddressLink {
      align-items: flex-start;
    }
  }
}
