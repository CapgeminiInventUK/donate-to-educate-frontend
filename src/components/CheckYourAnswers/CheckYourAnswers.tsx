import type { CheckYourAnswersProps } from '@/types/props';
import { checkYourAnswersDataMap } from '@/utils/formUtils';
import type { FC } from 'react';
import Button from '../Button/Button';
import styles from './CheckYourAnswers.module.scss';

const CheckYourAnswers: FC<CheckYourAnswersProps> = ({
  sections,
  formName,
  formData,
  setPageNumber,
}) => {
  const cyaData = checkYourAnswersDataMap(formName, formData);
  const handleClick = (page?: number): void => {
    if (!page || !setPageNumber) {
      return;
    }
    setPageNumber(page);
  };

  return sections.map((section, index) => (
    <div key={index} className={styles.container}>
      <h3 className={styles.header}>{section}</h3>
      {cyaData?.[section].map(({ field, value, page }, index) => (
        <div key={index} className={styles.row}>
          <div className={styles.field}>{field}</div>
          <div
            aria-label="value-cell"
            className={`${styles.value} ${field === 'Address' ? styles.addressCell : ''}`}
          >
            {value}
          </div>
          <div
            aria-label="change-link-cell"
            className={`${styles.changeLink} ${
              field === 'Address' ? styles.changeAddressLink : ''
            }`}
          >
            <Button
              theme="link"
              onClick={() => handleClick(page)}
              text="Change"
              ariaLabel="change"
            />
          </div>
        </div>
      ))}
    </div>
  ));
};
export default CheckYourAnswers;
