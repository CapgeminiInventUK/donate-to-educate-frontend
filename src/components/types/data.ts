import { ErrorInfo } from 'react';

export interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
  errorInfo?: ErrorInfo;
}
