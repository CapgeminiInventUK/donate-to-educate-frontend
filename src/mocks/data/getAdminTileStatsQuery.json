{
  "getAdminTileStats": {
    "la": {
      "joined": 19,
      "notJoined": 163,
      "__typename": "LocalAuthorityStats"
    },
    "joinRequests": {
      "school": 29,
      "charity": 9,
      "__typename": "JoinRequestStats"
    },
    "registeredSchools": 26,
    "registeredCharities": 12,
    "__typename": "AdminStats"
  }
}
