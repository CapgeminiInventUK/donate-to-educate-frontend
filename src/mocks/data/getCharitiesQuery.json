{
  "getCharities": [
    {
      "id": "123456",
      "localAuthority": "West Sussex",
      "name": "Test",
      "address": "Test Address\nTE5 7ST\n",
      "about": "sadfasdfasd",
      "distance": null,
      "profile": null,
      "postcode": "TE5 7ST",
      "__typename": "Charity"
    },
    {
      "id": "654321",
      "localAuthority": "West Sussex",
      "name": "Test two",
      "address": "Test Address Two\nTE5 7ST\n",
      "about": "asdfasdfa",
      "distance": null,
      "profile": null,
      "postcode": "TE5 7ST",
      "__typename": "Charity"
    }
  ]
}
