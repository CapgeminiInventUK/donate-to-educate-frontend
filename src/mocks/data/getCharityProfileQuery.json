{
  "getCharityProfile": {
    "name": "Test",
    "id": "1",
    "localAuthority": "West Sussex",
    "postcode": "TE5 T00",
    "location": {
      "type": "Point",
      "coordinates": [1.327269, 51.962367],
      "__typename": "Point"
    },
    "header": {
      "phone": "07777777777",
      "email": "test@test.com",
      "website": "www.test.com",
      "address": "Test",
      "__typename": "CharityProfileHeader"
    },
    "about": "Test charity has pre-loved school products to help children thrive at school.\n\n",
    "request": {
      "items": "{\"0\":[\"Coats\",\"Trousers\"]}",
      "whatToExpect": "View the products we have in stock. While we update our stock levels regularly, they may change daily.",
      "actionText": "Once we have your request for the products you need, we'll contact you to arrange the next steps as soon as we can.",
      "productTypes": [0],
      "__typename": "ProfileItems"
    },
    "donate": {
      "items": "{\"5\":[\"Internet access\",\"Laptops\",\"Computer software\"]}",
      "whatToExpect": "View the products we need. When you select 'donate', you can tell us how you can help.",
      "actionText": "Once we have your message about the products you can donate, we'll contact you to arrange the next steps as soon as we can.",
      "productTypes": [5],
      "__typename": "ProfileItems"
    },
    "excess": {
      "items": "{\"0\":[],\"1\":[],\"3\":[],\"5\":[\"Internet access\"]}",
      "whatToExpect": "View the products we have too much of, take it from us and share it with people who need it.",
      "actionText": "Once we know what extra stock you can take from us, we'll contact you to arrange the next steps as soon as we can.",
      "productTypes": [0, 1, 3, 5],
      "__typename": "ProfileItems"
    },
    "__typename": "CharityProfile"
  }
}
