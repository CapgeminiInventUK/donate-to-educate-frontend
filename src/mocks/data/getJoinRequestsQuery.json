{
  "getJoinRequests": [
    {
      "id": "123456",
      "name": "Test School",
      "localAuthority": "West Sussex",
      "type": "school",
      "requestTime": 1711394605001,
      "status": "NEW",
      "email": "test.school@test.com",
      "school": "Ormiston Six Villages Academy - PO20 3UE",
      "jobTitle": "Test",
      "phone": "07777777777",
      "charityName": null,
      "charityAddress": null,
      "aboutCharity": null,
      "urn": null,
      "postcode": null,
      "__typename": "JoinRequest"
    },
    {
      "id": "123457",
      "name": "Test Charity",
      "localAuthority": "Hammersmith and Fulham",
      "type": "charity",
      "requestTime": 1711394803223,
      "status": "NEW",
      "email": "test.charity@test.com",
      "school": "",
      "jobTitle": "Test",
      "phone": "07777777777",
      "charityName": "Test",
      "charityAddress": "Test\n          \n          Test\n          Test\n          Testttt\n          ",
      "aboutCharity": "Test",
      "urn": null,
      "postcode": null,
      "__typename": "JoinRequest"
    }
  ]
}
