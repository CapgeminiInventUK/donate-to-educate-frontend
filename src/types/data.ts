import { ErrorInfo, ReactNode } from 'react';
import {
  ButtonProps,
  CheckYourAnswersProps,
  CheckboxProps,
  DropdownProps,
  ExternalLinkProps,
  FormIntroPageProps,
  RadioGroupProps,
  SummaryProps,
  TextAreaProps,
  TextInputProps,
} from './props';
import Paths from '@/config/paths';
import { CharityProfileHeader, Point, ProfileItems, SchoolProfileHeader } from './api';

export interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
  errorInfo?: ErrorInfo;
  location?: string;
}

export interface CarouselItem {
  title: string;
  image: string;
  colour: 'lightBlue' | 'midBlue' | 'darkBlue';
}

export interface FormComponent {
  componentType: ComponentType;
  componentData?: ComponentDataPropsType;
  formComponentLink?: ExternalLinkProps;
  classNameSuffix?: string;
  field?: string;
}

export type ComponentDataPropsType =
  | TextInputProps
  | RadioGroupProps
  | CheckboxProps
  | TextAreaProps
  | DropdownProps
  | FormIntroPageProps
  | CheckYourAnswersProps
  | SummaryProps;

export interface FormTemplate {
  formComponents: FormComponent[];
  header?: ReactNode;
  subHeader?: ReactNode;
  secondaryHeader?: ReactNode;
  infoText?: string;
  infoTextTwo?: string;
  logo?: JSX.Element;
  footerLogo?: JSX.Element;
  isUnhappyPath?: boolean;
  summaryPageBg?: SummaryPageColour;
  formComponentInternalLink?: ButtonProps;
  onLocalAuthorityRegisterRequest?: () => void;
  isDeclarationPage?: boolean;
}

export enum SummaryPageColour {
  WHITE = 'white',
  BLUE = 'blue',
}

export interface FormMeta {
  page?: number;
  field?: string;
  section?: FormSections;
}

export interface FormDataItem {
  field: string;
  value: string | boolean;
  page?: number;
  section?: FormSections;
  fullValue?: Record<string, string | boolean>;
}

export enum FormSections {
  CHARITY_SECTION = 'Your charity or volunteer group',
  YOUR_DETAILS_SECTION = 'Your details',
  DECLARATION = 'Declaration',
}

export enum FormNames {
  CHARITY = 'Sign up charity',
  SCHOOL = 'Sign up school',
  AUTHORITY = 'Sign up Authority',
}

export enum FormErrors {
  EMAIL_ERROR_MESSAGE = 'Enter the email address in the correct format, like team@donatetoeducate.org.uk',
  PHONE_ERROR_MESSAGE = 'Enter a UK phone number in the correct format, like 07123456789',
  POSTCODE_ERROR_MESSAGE = 'Enter the postcode in the correct format, like LL70 9DJ',
  POSTCODE_NOT_FOUND = 'Postcode not found',
  TEXTAREA_MAX_LENGTH = 'Enter no more than 1000 characters',
}

export interface FormState {
  firstName: string;
  lastName: string;
  jobTitle: string;
  department: string;
  email: string;
  phone: string;
  notes: string;
}

export interface RequestFormState {
  name: string;
  email: string;
  phone: string;
  message: string;
  who: string;
  connection?: string;
}

export enum ComponentType {
  TEXT = 'textInput',
  CHECKBOX = 'checkbox',
  RADIO = 'radio',
  TEXTAREA = 'textArea',
  DROPDOWN = 'dropdown',
  INTRO = 'intro',
  SCHOOL_NOT_FOUND = 'schoolNotFound',
  CYA = 'checkYourAnswers',
  SUMMARY = 'summaryPage',
}

export interface DropdownOption {
  value: string;
  label: string;
  name?: string;
  postcode?: string;
  registered?: boolean;
}

export interface FilterDropdownOption {
  value: string;
  label: string;
  data: DropdownOption;
}

export interface NavRoute {
  path: Paths;
  childNavRoutes?: NavRoute[];
}

export interface SubmittedFormData {
  name: string;
  email: string;
  jobTitle: string;
  school?: string | null;
  phone?: string | null;
  charityName?: string | null;
  charityAddress?: string | null;
  aboutCharity?: string | null;
  message?: string;
  urn?: string;
  postcode?: string;
}

export interface SchoolOrCharityTableData {
  name: string;
  status: string;
  id?: string;
  urn?: string;
  joinRequestName?: string;
  jobTitle?: string | null;
  email?: string;
  phone?: string | null;
  charityAddress?: string | null;
  aboutCharity?: string | null;
}

export type myStageType = 'deciding' | 'approved' | 'denied';

export enum StageState {
  VIEW = 'view_requests',
  REMOVE = 'remove_school',
  APPROVE_SCHOOL = 'request_approval_school',
  APPROVE_CHARITY = 'request_approval_charity',
  REMOVED = 'removed',
}

export interface SchoolOrCharityProperties {
  id: string;
  name: string;
  la: string;
  user: RequestUser;
  urn?: string;
  charity?: { charityAddress: string; aboutCharity: string };
}

export interface RequestUser {
  name: string;
  title: string;
  email: string;
  phone: string;
}

export interface CustomAttributes {
  'custom:type': string;
  'custom:institution': string;
  'custom:institutionId': string;
}

export type AccountType = 'admin' | 'charity' | 'school' | 'localAuthority';

export interface Banner {
  phone?: string;
  email?: string;
  website?: string;
  uniformPolicy?: string;
  address?: string;
}

export type ResultType = 'declined' | 'approved';

export interface SignUpParameters {
  password: string;
  email: string;
  type: string;
  name: string;
  id: string;
}

export interface ConfirmSignUpParameters {
  email: string;
  code: string;
}

export type ItemsIconType = 'tick' | 'heart' | 'plus';

export type SectionsIconType =
  | 'Clothing and uniform'
  | 'Sports'
  | 'Art and music'
  | 'Study'
  | 'Toiletries'
  | 'Computing and technology';

export interface ItemList {
  name: SectionsIconType;
  items: string[];
}

export enum PillColours {
  GREEN = 'green',
  BLUE = 'blue',
  YELLOW = 'yellow',
  GREY = 'grey',
  RED = 'red',
  LIGHTBLUE = 'lightBlue',
}

export enum ActionTypes {
  REQUEST = 'request',
  DONATE = 'donate',
  EXCESS = 'excess',
}

export interface InstitutionProfile {
  name: string;
  id: string;
  localAuthority: string;
  postcode?: string | null;
  location?: Point | null;
  header?: CharityProfileHeader | SchoolProfileHeader | null;
  about?: string | null;
  request?: ProfileItems | null;
  donate?: ProfileItems | null;
  excess?: ProfileItems | null;
}

export interface RequestItemsTextContent {
  radioButtonLabels: string[];
  radioButtonValues: string[];
  buttonText: string;
  heading: string;
  subHeading: string;
  notesHeading: string;
  notesSubHeading: string;
}

export interface UserDetails {
  name: string;
  jobTitle: string;
  email: string;
  phone: string;
  institutionName?: string;
  id?: string;
  department?: string;
  firstName?: string;
  lastName?: string;
}

export interface ManageUserDetails {
  Name: string;
  Email: string;
  Phone: string;
  'Job title or role': string;
  Department?: string;
}

export enum InstitutionType {
  SCHOOL = 'school',
  CHARITY = 'charity',
}
