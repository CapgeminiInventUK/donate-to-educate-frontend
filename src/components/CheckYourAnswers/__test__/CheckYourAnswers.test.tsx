import { FormNames, FormSections } from '@/types/data';
import { render } from '@testing-library/react';
import CheckYourAnswers from '../CheckYourAnswers';
import userEvent from '@testing-library/user-event';
import styles from '../CheckYourAnswers.module.scss';

describe('Check your answers', () => {
  const sections = [FormSections.YOUR_DETAILS_SECTION];
  const formName = FormNames.SCHOOL;
  const formData = [
    {
      field: 'School',
      value: 'Test School - TE57 0NE',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: undefined,
      fullValue: {
        value: '125808',
        label: 'Test School - TE57 0NE',
        name: 'Test School',
        localAuthority: 'Test Auth',
        postcode: 'TE57 0NE',
        registered: false,
      },
    },
    {
      field: 'Job title or role',
      value: 'Title',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: 2,
    },
    {
      field: 'Email',
      value: 'test@test.com',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: 2,
    },
    {
      field: 'Phone',
      value: '+447777777777',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: 2,
    },
    {
      field: 'First name',
      value: 'First',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: 2,
    },
    {
      field: 'Last name',
      value: 'Last',
      section: FormSections.YOUR_DETAILS_SECTION,
      page: 2,
    },
  ];

  it('should render each section with a section header', () => {
    const Cya = render(
      <CheckYourAnswers sections={sections} formName={formName} formData={formData} />
    );
    expect(Cya).toMatchSnapshot();
  });

  it('should handle button click', async () => {
    const setPageNumber = vi.fn();
    const { getAllByRole } = render(
      <CheckYourAnswers
        sections={sections}
        formName={formName}
        formData={formData}
        setPageNumber={setPageNumber}
      />
    );
    const changeButtons = getAllByRole('button');
    await userEvent.click(changeButtons[0]);
    expect(setPageNumber).toHaveBeenCalledWith(2);
  });

  it('should not handle button click if no setPageNumber not passed', async () => {
    const setPageNumber = vi.fn();
    const { getAllByRole } = render(
      <CheckYourAnswers sections={sections} formName={formName} formData={formData} />
    );
    const changeButtons = getAllByRole('button');
    await userEvent.click(changeButtons[0]);
    expect(setPageNumber).not.toHaveBeenCalled();
  });

  it('should not handle button click if no page number', async () => {
    const setPageNumber = vi.fn();
    const { getAllByRole } = render(
      <CheckYourAnswers
        sections={sections}
        formName={formName}
        formData={formData}
        setPageNumber={setPageNumber}
      />
    );
    const changeButtons = getAllByRole('button');
    await userEvent.click(changeButtons[1]);
    expect(setPageNumber).not.toHaveBeenCalled();
  });
});

describe('CYA Charity Form', () => {
  it('should set address field properties when address field added', () => {
    const sections = [FormSections.CHARITY_SECTION];
    const formName = FormNames.CHARITY;
    const formData = [
      {
        field: 'Address line 1',
        value: 'Line 1',
        section: FormSections.CHARITY_SECTION,
        page: 4,
      },
      {
        field: 'Town',
        value: 'Town',
        section: FormSections.CHARITY_SECTION,
        page: 4,
      },
      {
        field: 'Postcode',
        value: 'PO5 7CD',
        section: FormSections.CHARITY_SECTION,
        page: 4,
      },
      {
        field: 'County',
        value: 'County',
        section: FormSections.CHARITY_SECTION,
        page: 4,
      },
    ];
    const setPageNumber = vi.fn();
    const { getAllByLabelText } = render(
      <CheckYourAnswers
        sections={sections}
        formName={formName}
        formData={formData}
        setPageNumber={setPageNumber}
      />
    );
    const valueCells = getAllByLabelText('value-cell');
    const changeLinkCells = getAllByLabelText('change-link-cell');
    expect(valueCells[0]).toHaveClass(styles.addressCell);
    expect(changeLinkCells[0]).toHaveClass(styles.changeAddressLink);
  });
});
