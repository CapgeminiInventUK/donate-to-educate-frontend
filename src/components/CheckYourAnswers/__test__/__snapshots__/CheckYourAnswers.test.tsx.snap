// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check your answers > should render each section with a section header 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_f0f2f5"
      >
        <h3>
          Your details
        </h3>
        <div
          class="_row_f0f2f5"
        >
          <div
            class="_field_f0f2f5"
          >
            Name
          </div>
          <div
            aria-label="value-cell"
            class="_value_f0f2f5 "
          >
            First Last
          </div>
          <div
            aria-label="change-link-cell"
            class="_changeLink_f0f2f5 "
          >
            <button
              aria-label="change"
              class="_link_f8f296  "
            >
              Change
            </button>
          </div>
        </div>
        <div
          class="_row_f0f2f5"
        >
          <div
            class="_field_f0f2f5"
          >
            School
          </div>
          <div
            aria-label="value-cell"
            class="_value_f0f2f5 "
          >
            Test School - TE57 0NE
          </div>
          <div
            aria-label="change-link-cell"
            class="_changeLink_f0f2f5 "
          >
            <button
              aria-label="change"
              class="_link_f8f296  "
            >
              Change
            </button>
          </div>
        </div>
        <div
          class="_row_f0f2f5"
        >
          <div
            class="_field_f0f2f5"
          >
            Job title or role
          </div>
          <div
            aria-label="value-cell"
            class="_value_f0f2f5 "
          >
            Title
          </div>
          <div
            aria-label="change-link-cell"
            class="_changeLink_f0f2f5 "
          >
            <button
              aria-label="change"
              class="_link_f8f296  "
            >
              Change
            </button>
          </div>
        </div>
        <div
          class="_row_f0f2f5"
        >
          <div
            class="_field_f0f2f5"
          >
            Email
          </div>
          <div
            aria-label="value-cell"
            class="_value_f0f2f5 "
          >
            test@test.com
          </div>
          <div
            aria-label="change-link-cell"
            class="_changeLink_f0f2f5 "
          >
            <button
              aria-label="change"
              class="_link_f8f296  "
            >
              Change
            </button>
          </div>
        </div>
        <div
          class="_row_f0f2f5"
        >
          <div
            class="_field_f0f2f5"
          >
            Phone
          </div>
          <div
            aria-label="value-cell"
            class="_value_f0f2f5 "
          >
            +447777777777
          </div>
          <div
            aria-label="change-link-cell"
            class="_changeLink_f0f2f5 "
          >
            <button
              aria-label="change"
              class="_link_f8f296  "
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_f0f2f5"
    >
      <h3>
        Your details
      </h3>
      <div
        class="_row_f0f2f5"
      >
        <div
          class="_field_f0f2f5"
        >
          Name
        </div>
        <div
          aria-label="value-cell"
          class="_value_f0f2f5 "
        >
          First Last
        </div>
        <div
          aria-label="change-link-cell"
          class="_changeLink_f0f2f5 "
        >
          <button
            aria-label="change"
            class="_link_f8f296  "
          >
            Change
          </button>
        </div>
      </div>
      <div
        class="_row_f0f2f5"
      >
        <div
          class="_field_f0f2f5"
        >
          School
        </div>
        <div
          aria-label="value-cell"
          class="_value_f0f2f5 "
        >
          Test School - TE57 0NE
        </div>
        <div
          aria-label="change-link-cell"
          class="_changeLink_f0f2f5 "
        >
          <button
            aria-label="change"
            class="_link_f8f296  "
          >
            Change
          </button>
        </div>
      </div>
      <div
        class="_row_f0f2f5"
      >
        <div
          class="_field_f0f2f5"
        >
          Job title or role
        </div>
        <div
          aria-label="value-cell"
          class="_value_f0f2f5 "
        >
          Title
        </div>
        <div
          aria-label="change-link-cell"
          class="_changeLink_f0f2f5 "
        >
          <button
            aria-label="change"
            class="_link_f8f296  "
          >
            Change
          </button>
        </div>
      </div>
      <div
        class="_row_f0f2f5"
      >
        <div
          class="_field_f0f2f5"
        >
          Email
        </div>
        <div
          aria-label="value-cell"
          class="_value_f0f2f5 "
        >
          test@test.com
        </div>
        <div
          aria-label="change-link-cell"
          class="_changeLink_f0f2f5 "
        >
          <button
            aria-label="change"
            class="_link_f8f296  "
          >
            Change
          </button>
        </div>
      </div>
      <div
        class="_row_f0f2f5"
      >
        <div
          class="_field_f0f2f5"
        >
          Phone
        </div>
        <div
          aria-label="value-cell"
          class="_value_f0f2f5 "
        >
          +447777777777
        </div>
        <div
          aria-label="change-link-cell"
          class="_changeLink_f0f2f5 "
        >
          <button
            aria-label="change"
            class="_link_f8f296  "
          >
            Change
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
